pkg/storage/chunk/cache/embeddedcache.go

```diff
@@ -0,0 +1,47 @@
+package cache
+
+import (
+	"flag"
+	"time"
+)
+
+const (
+	DefaultPurgeInterval = 1 * time.Minute
+)
+
+// EmbeddedcacheConfig represents in-process embedded cache config.
+// It can also be distributed, sharding keys across peers when run with microservices
+// or SSD mode.
+type EmbeddedcacheConfig struct {
+	Distributed bool          `yaml:"distributed,omitempty"`
+	Enabled     bool          `yaml:"enabled,omitempty"`
+	MaxSizeMB   int64         `yaml:"max_size_mb"`
+	MaxItems    int           `yaml:"max_items"`
+	TTL         time.Duration `yaml:"ttl"`
+
+	// PurgeInterval tell how often should we remove keys that are expired.
+	// by default it takes `DefaultPurgeInterval`
+	PurgeInterval time.Duration `yaml:"-"`
+
+	// distributed cache configs. Have no meaning if `Distributed=false`.
+	Ring       RingCfg `yaml:"ring,omitempty"`
```

Comment:
I don't think we can have the ring config on the cache config. Because groupcache globally shares state, there can only ever be one ring. When we want to add another cache, this becomes very confisuing.